File: backend/src/main/java/com/studyai/wellness/dto/DashboardDto.java
package com.studyai.wellness.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Data Transfer Object for Dashboard information.
 *
 * <p>This DTO aggregates user metrics, habits, and weekly progress
 * for the dashboard view.</p>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DashboardDto {

    /**
     * User information.
     */
    private UserDto user;

    /**
     * List of health/fitness metrics.
     */
    private List<MetricDto> metrics;

    /**
     * List of daily habits.
     */
    private List<HabitDto> habits;

    /**
     * Weekly progress data.
     */
    private WeeklyProgressDto weeklyProgress;
}

File: android/app/src/main/java/com/studyai/wellness/data/model/DashboardDto.kt
package com.studyai.wellness.data.model

data class DashboardDto(
    val user: UserDto,
    val metrics: List<MetricDto>,
    val habits: List<HabitDto>,
    val weeklyProgress: WeeklyProgressDto
)

data class MetricDto(
    val id: String,
    val title: String,
    val value: String,
    val subtitle: String? = null,
    val trend: String? = null,
    val color: String? = null
)

data class HabitDto(
    val id: String,
    val name: String,
    val icon: String,
    val completed: Boolean,
    val streak: Int
)

data class WeeklyProgressDto(
    val days: List<DayProgressDto>
)

data class DayProgressDto(
    val day: String,
    val value: Double,
    val completed: Boolean
)
